### PR TITLE
create time service

### PR DESCRIPTION
**Pull request by kosyfrances**
#7

### CODE REVIEW

**Review by kola-er — APPROVED** [review]
Looks good.

**Review by sundayguru — APPROVED** [review]
Looks good

File: common/utils.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    """
+
+    def yesterday(self):
+        return date.today() - timedelta(days=1)
+
+    def today(self):
+        return date.today()
+
+    def tomorrow(self):
+        return date.today() + timedelta(days=1)
```
Comment (nbyodai) [inline common/utils.py]: Although this looks good, I'm not sure it covers the use case very well. The idea behind the TimeService is that if a user puts in `tomorrow`, `yesterday`, `today`, `next tomorrow` we should have a way to parse that information and return the date object for our graphql to use as query filter.

If you look at your implementation, the only difference is the value being passed into `timedelta`. it also takes in a negative number so for where you've got
`date.today() - timedelta(days=1)` is the same as 
`date.today() + timedelta(days=-1)`
and today is same as `date.today() + timedelta(days=0)`

We only need one function
```
def get_date(day_arg=0):
  # Do the magic of conversion inside here
  return date.today() + timedelta(days=day_arg)
```
The TimeService would still need a good way to parse possible entries in order to get the `day_arg`
These are possible different types 
**number_days_type** e.g `5 days, 2 days`
**today_tomorrow_yesterday_type** e.g `today, tomorrow,  `
**day_name_type** e.g `monday , tuesday`
**next_type** e.g `next week, next monday, next wednesday`
**date_type** e.g `12th of March`

We would need another function to accurately determine the `day _arg` based on the type.

**Review by nbyodai — CHANGES_REQUESTED** [review]

File: common/utils.py (hunk 1 above)
Comment (nbyodai) [reply common/utils.py]: Do reach out if anything is unclear

**Comment by kosyfrances** [comment]
@andela-cobiora @kola-er @andela-snwuguru please review above changes.

File: common/utils.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            # In python, Monday is 0 and Sunday is 6
+            current_weekday_num = date.today().weekday()
+            day_index = days.index(day_arg)
+
+            if current_weekday_num == day_index:
+                # The weekday passed in is today
+                return day_to_num_dict['today']
+            elif current_weekday_num > day_index:
+                # It is a previous weekday
+                return 0 - (current_weekday_num - day_index)
```
Comment (nbyodai) [inline common/utils.py]: if you use `day_index - current_weekday_num` here, then you wouldn't have to use the `0 -` part

File: common/utils.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            else:
+                # It is a future weekday
+                return (day_index - current_weekday_num)
```
Comment (nbyodai) [inline common/utils.py]: Is the use of `()` here necessary?

File: tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import datetime
+import unittest
+
+from freezegun import freeze_time
+
+from common.utils import get_date
+
+
+class GetDateTest(unittest.TestCase):
```
Comment (nbyodai) [inline tests/test_utils.py:9]: 👍

**Review by nbyodai — CHANGES_REQUESTED** [review]

File: common/utils.py (hunk 3 above)
Comment (kosyfrances) [reply common/utils.py]: Just for clean code consistency sake.

**Comment by kosyfrances** [comment]
@andela-cobiora changes updated

**Review by nbyodai — APPROVED** [review]

**Pull request merged**